ibc/src/component/client.rs

```diff
@@ -352,44 +354,52 @@ impl Ics2Client {
 #[async_trait]
 pub trait View: StateExt {
     async fn put_client_counter(&mut self, counter: ClientCounter) {
-        self.put_domain("ibc/ics02-client/client_counter".into(), counter)
-            .await;
+        self.put_domain("ibc_client_counter".into(), counter).await;
     }
     async fn client_counter(&self) -> Result<ClientCounter> {
-        self.get_domain("ibc/ics02-client/client_counter".into())
+        self.get_domain("ibc_client_counter".into())
             .await
             .map(|counter| counter.unwrap_or(ClientCounter(0)))
     }
-    async fn put_client_data(&mut self, data: ClientData) {
+
+    async fn put_client(&mut self, client_id: &ClientId, client_state: AnyClientState) {
+        self.put_proto(
+            format!("{}/clients/{}/clientType", COMMITMENT_PREFIX, client_id).into(),
+            client_state.client_type().as_str().to_string(),
+        )
+        .await;
+
         self.put_domain(
-            format!(
-                "ibc/ics02-client/clients/{}",
-                hex::encode(data.client_id.as_bytes())
-            )
-            .into(),
-            data,
+            format!("{}/clients/{}/clientState", COMMITMENT_PREFIX, client_id).into(),
+            client_state,
         )
         .await;
     }
-    async fn get_client_data(&self, client_id: &ClientId) -> Result<ClientData> {
-        let client_data = self
-            .get_domain(
-                format!(
-                    "ibc/ics02-client/clients/{}",
-                    hex::encode(client_id.as_bytes())
-                )
-                .into(),
-            )
+
+    async fn get_client_type(&self, client_id: &ClientId) -> Result<ClientType> {
+        let client_type_str: String = self
+            .get_proto(format!("{}/clients/{}/clientType", COMMITMENT_PREFIX, client_id).into())
+            .await?
+            .ok_or(anyhow::anyhow!("client not found"))?;
+
+        ClientType::from_str(&client_type_str).map_err(|_| anyhow::anyhow!("invalid client type"))
+    }
+
+    async fn get_client_state(&self, client_id: &ClientId) -> Result<AnyClientState> {
+        let client_state = self
+            .get_domain(format!("{}/clients/{}/clientState", COMMITMENT_PREFIX, client_id).into())
             .await?;
 
-        client_data.ok_or(anyhow::anyhow!("client not found"))
+        client_state.ok_or(anyhow::anyhow!("client not found"))
     }
 
     async fn get_verified_heights(&self, client_id: &ClientId) -> Result<Option<VerifiedHeights>> {
         self.get_domain(
             format!(
-                "ibc/ics02-client/clients/{}/verified_heights",
-                hex::encode(client_id.as_bytes())
+                // NOTE: this is an implementation detail of the Penumbra ICS2 implementation, so
+                // it's not in the same path namespace.
+                "penumbra_verified_heights/{}/verified_heights",
```

Comment:
non-blocking nit: maybe these should be `penumbra_ibc/verified_heights/`, so that they're all under the same path namespace, even though it's distinct from the `ibc` namespace?